app/controllers/CancelDeclarationController.scala

```diff
@@ -54,13 +63,30 @@ class CancelDeclarationController @Inject()(
         (formWithErrors: Form[CancelDeclaration]) =>
           Future.successful(BadRequest(cancelDeclarationPage(formWithErrors))),
         form => {
+          auditService.auditAllPagesUserInput(Json.toJson(form).as[JsObject])
```

Comment:
Could change this method to accept a generic argument and an implicit Writes to hide some of the json boilerplate

Reply:
Sure. I would like to do this with you.